deltachat-ios/Controller/Settings/NotificationsViewController.swift

```diff
@@ -0,0 +1,145 @@
+        cell.tag = CellTags.systemSettings.rawValue
+        cell.textLabel?.text = String.localized("system_settings")
+        cell.accessoryType = .disclosureIndicator
+        return cell
+    }()
+
+    private lazy var sections: [SectionConfigs] = {
+        let preferencesSection = SectionConfigs(
+            headerTitle: nil,
+            footerTitle: String.localized("pref_mention_notifications_explain"),
+            cells: [notificationsCell, mentionsCell]
+        )
+        let systemSettingsSection = SectionConfigs(
+            headerTitle: nil,
+            footerTitle: String.localized("system_settings_notify_explain_ios"),
+            cells: [systemSettingsCell]
+        )
+        return [preferencesSection, systemSettingsSection]
+    }()
+
+    init(dcAccounts: DcAccounts) {
+        self.dcContext = dcAccounts.getSelected()
+        self.dcAccounts = dcAccounts
+        super.init(style: .insetGrouped)
+        hidesBottomBarWhenPushed = true
+    }
+
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = String.localized("pref_notifications")
+        tableView.rowHeight = UITableView.automaticDimension
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        updateCells()
+    }
+
+    // MARK: - UITableViewDelegate + UITableViewDatasource
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return sections.count
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sections[section].cells.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        return sections[indexPath.section].cells[indexPath.row]
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return sections[section].headerTitle
+    }
+
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        return sections[section].footerTitle
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let cell = tableView.cellForRow(at: indexPath), let cellTag = CellTags(rawValue: cell.tag) else { safe_fatalError(); return }
+        tableView.deselectRow(at: indexPath, animated: false)
+
+        switch cellTag {
+        case .systemSettings:
+            let urlString = if #available(iOS 16, *) {
+                UIApplication.openNotificationSettingsURLString
+            } else if #available(iOS 15.4, *) {
+                UIApplicationOpenNotificationSettingsURLString
+            } else {
+                UIApplication.openSettingsURLString
+            }
```

Comment:
I find this if-assignment-thinkie kind of ... confusing but maybe I'm an old man yelling at clouds

Reply:
it is pretty common nowadays, known eg. from rust. i think, it is a handy tool - and avoids having mutable vars

Reply:
As I said: I'm an old man yelling at clouds